### PR TITLE
Feature: wallet rendering. Create uniform Credentials/Presentations from JWT / LD

### PR DESCRIPTION
**Pull request by nklomp**
Extracted and refactored from PEX

### CODE REVIEW

File: packages/ssi-types/src/types/vc.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { PresentationSubmission } from './pex'
 import { IProofPurpose, IProofType } from './did'
 
+type AdditionalProperties = Record<string, any>
+
 export interface ICredential {
```
Comment (sksadjad) [inline packages/ssi-types/src/types/vc.ts:6]: we might need to add a couple of optional values here:
https://www.w3.org/TR/vc-data-model/#nontransferable-property
credentialStatus
https://www.w3.org/TR/vc-data-model/#dfn-evidence
https://www.w3.org/TR/vc-data-model/#terms-of-use
https://www.w3.org/TR/vc-data-model/#refreshing
https://www.w3.org/TR/vc-data-model/#data-schemas

File: packages/ion-did-provider/__tests__/ion-did-provider.test.ts [hunk 2]
```diff
@@ -214,7 +214,7 @@ const createIdentifierOpts = {
     {
       kid: 'did1-test',
       purposes: [IonPublicKeyPurpose.Authentication, IonPublicKeyPurpose.AssertionMethod],
-    },
+      },
```
Comment (sksadjad) [inline packages/ion-did-provider/__tests__/ion-did-provider.test.ts:217]: prettier should pick this up

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_chapi.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (sksadjad) [inline packages/ssi-types/__tests__/dif_pe_examples/vp/vp_chapi.json:1]: since we don't support/use this, remove it

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_did_comm.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (sksadjad) [inline packages/ssi-types/__tests__/dif_pe_examples/vp/vp_did_comm.json:1]: since we don't support/use this, remove it

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_nested_submission.json [hunk 5]
```diff
@@ -0,0 +1,152 @@
+{
```
Comment (sksadjad) [inline packages/ssi-types/__tests__/dif_pe_examples/vp/vp_nested_submission.json:1]: since we don't support/use this, remove it

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_open_id_connect.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (sksadjad) [inline packages/ssi-types/__tests__/dif_pe_examples/vp/vp_open_id_connect.json:1]: since we don't support/use this, remove it

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_ps.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (sksadjad) [inline packages/ssi-types/__tests__/dif_pe_examples/vp/vp_ps.json:1]: since we don't support/use this, remove it

File: packages/ssi-types/src/types/vc.ts [hunk 8]
```diff
@@ -66,7 +69,7 @@ export interface ICredentialStatus {
 export interface IIssuer {
   id: string
 
-  [x: string]: unknown
+  [x: string]: any
```
Comment (sksadjad) [inline packages/ssi-types/src/types/vc.ts:72]: why the change to any?

File: packages/ssi-types/src/types/vc.ts [hunk 9]
```diff
@@ -90,57 +95,90 @@ export interface WrappedVerifiableCredential {
...
+   * Wrapped Verifiable Credentials belonging to the Presentation
+   */
   vcs: WrappedVerifiableCredential[]
 }
 
-export enum VerifiableDataExchangeType {
-  JSONLD,
-  JWT_ENCODED,
-  JWT_DECODED,
+export enum OriginalType {
```
Comment (sksadjad) [inline packages/ssi-types/src/types/vc.ts:144]: suggest to revert this one. `OriginalType` is too general

**Review by sksadjad — CHANGES_REQUESTED** [review]
added some small remarks.
for additional properties (the first remark on vc.ts, we can create a ticket because I think we might need some extra checks for that)
and also you're making lots of changes regarding the `CHANGELOG.md` files

File: packages/ssi-types/src/types/vc.ts (hunk 1 above)
Comment (nklomp) [reply packages/ssi-types/src/types/vc.ts:6]: Yeah, will add these later as part of wallet rendering when we encounter them.  Not really in the scope of this PR

File: packages/ssi-types/src/types/vc.ts (hunk 8 above)
Comment (nklomp) [reply packages/ssi-types/src/types/vc.ts:72]: Because you cannot simply access nested properties when it is unknown.

credentialSubject.mainProperty.subProperty

File: packages/ssi-types/src/types/vc.ts (hunk 9 above)
Comment (nklomp) [reply packages/ssi-types/src/types/vc.ts:144]: Problem is that it can be used for Credentials and Presentations and is scoped to a Wrapped VC or Wrapped VP, which also has an original field

I don't really have a better name tbh especially since it is clear from the context

File: packages/ssi-types/src/types/vc.ts (hunk 1 above)
Comment (sksadjad) [reply packages/ssi-types/src/types/vc.ts:6]: https://sphereon.atlassian.net/browse/VDX-143

File: packages/ion-did-provider/__tests__/ion-did-provider.test.ts (hunk 2 above)
Comment (nklomp) [reply packages/ion-did-provider/__tests__/ion-did-provider.test.ts]: Hmzz, had to do with a revert of a commit. Fixed

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_did_comm.json (hunk 4 above)
Comment (nklomp) [reply packages/ssi-types/__tests__/dif_pe_examples/vp/vp_did_comm.json]: removed and dir name changed

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_nested_submission.json (hunk 5 above)
Comment (nklomp) [reply packages/ssi-types/__tests__/dif_pe_examples/vp/vp_nested_submission.json]: removed and dir name changed

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_open_id_connect.json (hunk 6 above)
Comment (nklomp) [reply packages/ssi-types/__tests__/dif_pe_examples/vp/vp_open_id_connect.json]: removed and dir name changed

File: packages/ssi-types/__tests__/dif_pe_examples/vp/vp_ps.json (hunk 7 above)
Comment (nklomp) [reply packages/ssi-types/__tests__/dif_pe_examples/vp/vp_ps.json]: removed and dir name changed
